Backport #10231: [Qt] Reduce a significant cs_main lock freeze

codablock opened the pull request:
This is a backport of https://github.com/bitcoin/bitcoin/pull/10231, which fixes the lagging UI currently seen in testnet when large blocks are getting processed.

UdjinM6 commented:
Needs to be rebased/cherry-picked on top of 12.2, otherwise postponing till 12.2.1

codablock commented:
Rebased and changed base to 12.2.

To get this into the develop branch, I'd suggest to later merge v0.12.2.x into develop.
But for the future, we should maybe think about a better way to handle such cases. For example, create all PRs with develop as base and then manually cherry pick into the release branches. I think Bitcoin already has an established workflow with some helper scripts for this.
